lumicks/pylake/tests/test_fdensemble.py

```diff
@@ -16,6 +15,83 @@ def make_mock_fd(force, distance, start=0):
     return fd
 
 
+def test_align_distance_simple():
+    force = np.hstack((np.ones(50), np.arange(1, 50)))
+    distance = np.arange(1, 100)
+    fd1 = make_mock_fd(force=force, distance=distance, start=0)
+    fd2 = make_mock_fd(force=force + 5, distance=distance, start=0)
+    fd3 = make_mock_fd(force=force + 15, distance=distance, start=0)
+    aligned = align_force_simple({"fd1": fd1, "fd2": fd2, "fd3": fd3}, distance_range=50)
+
+    for fd in aligned.values():
+        assert np.allclose(fd.d.data, distance)
+        assert np.allclose(fd.f.data, force)
+
+
+def test_distance_force_simple():
+    force = np.hstack((np.ones(50), np.arange(1, 52)))
+    distance = np.arange(1.0, 102.0)
+    fd1 = make_mock_fd(force=force, distance=distance, start=0)
+    fd2 = make_mock_fd(force=force, distance=distance + 4.0, start=0)
+    fd3 = make_mock_fd(force=force, distance=distance + 2.0, start=0)
+    aligned = align_distance_simple({"fd1": fd1, "fd2": fd2, "fd3": fd3}, distance_range=50)
+
+    for fd in aligned.values():
+        assert np.allclose(fd.d.data, distance)
+        assert np.allclose(fd.f.data, force)
+
+
+def test_align_all_simple():
+    force = np.hstack((np.ones(50), np.arange(1, 52)))
+    distance = np.arange(1.0, 102.0)
+    fd1 = make_mock_fd(force=force, distance=distance, start=0)
+    fd2 = make_mock_fd(force=force + 5.0, distance=distance + 4.0, start=0)
+    fd3 = make_mock_fd(force=force + 15.0, distance=distance + 2.0, start=0)
+    aligned = align_fd_simple({"fd1": fd1, "fd2": fd2, "fd3": fd3}, 50, 50)
+
+    for fd in aligned.values():
+        assert np.allclose(fd.f.data, force)
+        assert np.allclose(fd.d.data, distance)
+
+
+def test_non_constant_rate_fd_alignment_simple():
+    """Tests what happens when we try to align an Fd curve with a non-constant pulling rate"""
```

Comment:
👍

Reply:
Not sure if you commit changes here?
What I really want to know here is: `what happens when we try to align an Fd curve with a non-constant pulling rate`?

Reply:
Ah yes, good point! I'll try and be more descriptive in my tests.
``` Tests what happens when we try to align an Fd curve with a non-constant pulling rate. A non-constant pulling rate would lead to a non-linear relation when just looking at either force or distance, rather than looking at them at the same time. Here we test whether the alignment actually operates on f,d rather than on channels independently"""```